manifests/base/gitops-service-argocd/base/argo-cd-clusterrole.yaml

```diff
@@ -0,0 +1,15 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: appstudio-openshift-gitops-argocd-application-controller
+rules:
```

Comment:
Something that occurred to me while reviewing the story: Rather than adding * * * cluster role, it likely makes more sense to scope Argo CD to only deploying a specific set of resources.

I think we can copy the list of resources from here:
https://github.com/redhat-appstudio/managed-gitops/blob/753e58188a55a7627f056bb8c2117ebdc7af69e9/manifests/base/gitops-service-argocd/base/argo-cd.yaml#L10

The idea is that this is a security defense-in-depth, to make sure that Argo CD only has write accesses to the reasons that we want it to. The list defined in the file above is the current list of resources that Argo CD can deploy.

Reply:
i tried this but the tests failed.  I added the CR/CRBs for the Argo CD server as well, taken from the roles and bindings created by the gitops operator.  The tests passed after this.  WDYT?